pkg/packagekit/updates.jsx

```diff
@@ -513,116 +513,106 @@ class RestartServices extends React.Component {
+                    const percent = props.Percentage.v;
+                    let remain = -1;
+                    if ("RemainingTime" in props)
+                        remain = props.RemainingTime.v;
+
+                    const pfields = (packageId || props.LastPackage.v).split(";");
+                    // status: see PK_STATUS_* at https://github.com/PackageKit/PackageKit/blob/main/lib/packagekit-glib2/pk-enum.h
+                    setActions(prevActions => [...prevActions, {
+                        status: status || props.Status.v,
+                        package: pfields[0] + " " + pfields[1] + " (" + pfields[2] + ")"
+                    }]);
+
+                    const log = document.getElementById("update-log");
+                    let atBottom = false;
+                    if (log) {
+                        if (log.scrollHeight - log.clientHeight <= log.scrollTop + 2)
+                            atBottom = true;
+                    }
 
-    componentDidMount() {
-        this._mounted = true;
-        const transactionPath = this.props.transaction;
-
-        PK.watchTransaction(transactionPath, {
-            Package: (info, packageId) => {
-                const pfields = packageId.split(";");
-
-                // small timeout to avoid excessive overlaps from the next PackageKit progress signal
-                PK.call(transactionPath, "org.freedesktop.DBus.Properties", "GetAll", [PK.transactionInterface], { timeout: 500 })
-                        .done(reply => {
-                            if (this._mounted === false)
-                                return;
-
-                            const percent = reply[0].Percentage.v;
-                            let remain = -1;
-                            if ("RemainingTime" in reply[0])
-                                remain = reply[0].RemainingTime.v;
-                            // info: see PK_STATUS_* at https://github.com/PackageKit/PackageKit/blob/main/lib/packagekit-glib2/pk-enum.h
-                            const newActions = this.state.actions.slice();
-                            newActions.push({ status: info, package: pfields[0] + " " + pfields[1] + " (" + pfields[2] + ")" });
-
-                            const log = document.getElementById("update-log");
-                            let atBottom = false;
-                            if (log) {
-                                if (log.scrollHeight - log.clientHeight <= log.scrollTop + 2)
-                                    atBottom = true;
-                            }
-
-                            this.setState({
-                                actions: newActions,
-                                percentage: percent <= 100 ? percent : 0,
-                                timeRemaining: remain > 0 ? remain : null
-                            });
+                    setPercentage(percent <= 100 ? percent : 0);
+                    setTimeRemaining(remain > 0 ? remain : null);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17517-20220705-092548-cbb7d34f-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)